### PR TITLE
patch .so attempt 2

### PR DESCRIPTION
**Pull request by jjsjann123**
rpath should be a concatenated string, previous patch ended up overwriting it

### CODE REVIEW

**Comment by jjsjann123** [comment]
Previous one #737  is a no-go. I should have verified it before merging that one :face_exhaling:

File: setup.py
Comment (zasdfgbnm) [inline setup.py:303]: Why?

Comment (zasdfgbnm) [reply setup.py:303]: If some node is failing, should we just set `MAX_JOBS` for that node? 24 jobs is such a small number. On a 12 core CPU, there will be 24 threads. And we have pretty much nodes with > 12 cores.

**Review by zasdfgbnm — APPROVED** [review]
Stamp to unblock, but I don't agree with the max jobs change. Please do it in a separate PR so we can discuss.

File: setup.py
Comment (zasdfgbnm) [reply setup.py:303]: I think my suggestion for this is: for a machine that people use daily, people can just set `MAX_JOBS` manually in their `bashrc` and use whatever jobs they want so it does not really matter for this case. But when someone needs to go to a new environment (especially when it is inside a docker container), for example some shared node in company, usually in that environment, the `MAX_JOBS` will not be set. We should make the cap here larger so in our common machines, the compilation speed is maximum without manually setting `MAX_JOBS`. It is bad experience if everytime when you grab a node, you need to do a `lscpu` and check what is the cpu number and set `MAX_JOBS` manually.

Comment (zasdfgbnm) [reply setup.py:303]: I don't mind merging this PR as is, but if this is the case, please do open a separate PR for more discussion. Also, should this be the following?
```suggestion
        max_jobs = os.getenv("MAX_JOBS", str(min(multiprocessing.cpu_count(), 24)))
```

Comment (jjsjann123) [reply setup.py:303]: well, I guess we don't have to. it's killing our a100 node on nightly container.

**Pull request merged**
